Automattic/jetpack
# E2E tests: Introduce steps in tests

#### Changes proposed in this Pull Request:

* Introduce `step`s to split up the tests in more meaningful pieces.
* use `reporter.startStep` & `reporter.endStep` to capture steps in Allure reports
* rename `connection.test.js` to `re-connection.test.js` for clarity

#### Testing instructions:
* Make sure tests are green
* check https://jetpack-e2e-dashboard.herokuapp.com/pr-15553 and make sure suites does have steps in test details

#### Proposed changelog entry for your changes:

* n/a
